lua/nvim-tree/renderer/builder.lua

```diff
@@ -149,7 +143,15 @@ function Builder:_build_folder(node)
   end
 
   local foldername_hl = "NvimTreeFolderName"
-  if vim.tbl_contains(self.special_files, node.absolute_path) or vim.tbl_contains(self.special_files, node.name) then
+  if node.link_to and self.symlink_destination then
+    local arrow = icons.i.symlink_arrow
+    local link_to = utils.path_relative(node.link_to, core.get_cwd())
+    foldername = foldername .. arrow .. link_to
+    foldername_hl = "NvimTreeSymlinkFolderName"
+    icon_hl = "NvimTreeSymlinkFolderName"
```

Comment:
```suggestion
```

Let's make this consistent with regular folders, which use FolderIcon instead of FolderName.

In the future `NvimTreeOpenedSymlinkFolderIcon` and `NvimTreeClosedSymlinkFolderIcon` may be added.